src/test/java/com/linecorp/armeria/client/manipulating/HttpHeaderManipulatingClientBuilderTest.java

```diff
@@ -0,0 +1,224 @@
+        HttpRequest request = newRequest();
+        Client client = newBuilder().buildClient(dummyDelegate);
+        client.execute(dummyContext, request);
+        assertEquals(3, request.headers().size());
+    }
+
+    @Test
+    public void testAdd() throws Exception {
+        HttpRequest request = newRequest();
+        Client client = newBuilder()
+            .add(HEADER_1, "a1")
+            .add(HEADER_0, "a0")
+            .buildClient(dummyDelegate);
+
+        client.execute(dummyContext, request);
+        HttpHeaders headers = request.headers();
+        
+        List<String> value0 = headers.getAll(HEADER_0);
+        List<String> value1 = headers.getAll(HEADER_1);
+
+        assertEquals("[v0, a0]", value0.toString());
+        assertEquals("[a1]", value1.toString());
+    }
+
+    @Test
+    public void testSet() throws Exception {
+        HttpRequest request = newRequest();
+        Client client = newBuilder()
+                .add(HEADER_0, "a0").add(HEADER_0, "a00").add(HEADER_0, "a000")
+                .set(HEADER_0, "s0") // should replace all entries
+                .set(HEADER_1, "s1") // should set value of non-existing header
+                .set(HEADER_1, value -> value + "!!") // can read old value and change it
+                .set(HEADER_2, "s2")
+                .set(HEADER_2, value -> "") // should ignore empty value
+                .set(HEADER_2, value -> null) // should ignore null value
+                .set(HEADER_3, value -> value == null ? "s3" : "BUG!") // should be called with null
+                .buildClient(dummyDelegate);
+
+        client.execute(dummyContext, request);
+        HttpHeaders headers = request.headers();
+        
+        List<String> value0 = headers.getAll(HEADER_0);
+        List<String> value1 = headers.getAll(HEADER_1);
+        List<String> value2 = headers.getAll(HEADER_2);
+        List<String> value3 = headers.getAll(HEADER_3);
+
+        assertEquals("[s0]", value0.toString());
+        assertEquals("[s1!!]", value1.toString());
+        assertEquals("[s2]", value2.toString());
+        assertEquals("[s3]", value3.toString());
+    }
+
+    @Test
+    public void testRemove() throws Exception {
+        HttpRequest request = newRequest();
+        request.headers().add(HEADER_0, "a0")
+                         .set(HEADER_1, "v1")
+                         .set(X_HEADER_1, "x1")
+                         .set(X_HEADER_2, "x2")
+                         .set(X_HEADER_3, "x3");
+        
+        Client client = newBuilder()
+                .remove(HEADER_0) // should remove all entries
+                .remove((header, value) -> header.toString().startsWith("x-")) // conditional remove
+                .add(X_HEADER_3, "X3")
+                .buildClient(dummyDelegate);
+
+        client.execute(dummyContext, request);
+        HttpHeaders headers = request.headers();
+
+        assertEquals(null, headers.get(HEADER_0));
```

Comment:
Please use `assertNull(headers.get(HEADER_0))` or `assertFalse(headers.contains(HEADER_0))` instead of comparing `null` and return value in `assertEquals`.